service/controller/v13/resource/cloudformation/create.go

```diff
@@ -56,6 +57,18 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 			return microerror.Mask(err)
 		}
 
+		if r.encrypterBackend == encrypter.VaultBackend {
+			outputs, _, err := sc.CloudFormation.DescribeOutputsAndStatus(*stackInput.StackName)
```

Comment:
Well, we make the request here anyway again. We already made the request in `GetCurrentState`. IMO we should just use the information we have available already.

Reply:
Please correct me if I'm wrong but AIUI, in this case, the previous call to `GetCurrentState` would not have returned any info about the stack outputs because it was not created yet. We query the outputs right after the creation.

Reply:
I see the issue. Did you test this already? I would expect this to not work at all because right after the creation the outputs are not yet available. We do this to get the ARNs. Isn't there another more lightway way of fetching these information? Why do we store the ARNs in the outputs anyway?

Reply:
I've tested but there seems to be a race condition, even if the outputs were available the workers might be already up and the role could be not in place by the time the decryption is required. 

I'll try to find another way, as you suggest we can determine the role ARNs just knowing the account id and the cluster id, will ping you when ready.

Reply:
@xh3b4sd done, now we have functions to determine role ARNs without querying the API, PTAL.